onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -383,8 +383,6 @@ Status CUDAExecutionProvider::Sync() const {
 }
 
 Status CUDAExecutionProvider::OnRunStart() {
-  // always set CUDA device when session::Run() in case it runs in a worker thread
-  CUDA_RETURN_IF_ERROR(cudaSetDevice(GetDeviceId()));
```

Comment:
should we keep it? there may be other cuda component other than stream depends on current device id, like cudnn

Reply:
This change is essentially moving cudaSetDevice() from OnRunStart() to 

DeviceStreamCollectionHolder device_stream_collection_holder(session_state_.get());

there are only a few lines of codes between these 2 places (in InferenceSession::Run()) and no cuda related operation involved. So I think deletion here is safe.